File: src/IFWEMS.Application/Cases/CaseModels.cs
using IFWEMS.Domain.Entities;

namespace IFWEMS.Application.Cases;

public record CreateCaseRequest(CaseType CaseType, Guid OrgUnitId, string Title, string? Description, decimal? AmountInvolved, Guid? SupplierId, Guid? ContractId);

public record CaseDto(
    Guid Id, string CaseNumber, CaseType CaseType, CaseStatus Status, Guid OrgUnitId,
    string Title, string? Description, decimal? AmountInvolved, decimal RecoverableAmount,
    decimal RecoveredAmount, DateTime CreatedAtUtc);

public record ChangeCaseStatusRequest(CaseStatus NewStatus, string? Reason);

File: src/IFWEMS.Client/src/app/core/models/case.models.ts
export type CaseType = 'IrregularExpenditure' | 'FruitlessWasteful' | 'UnauthorisedExpenditure' | 'PotentialNonCompliance';
export type CaseStatus =
  | 'Draft'
  | 'UnderAssessment'
  | 'UnderInvestigation'
  | 'Determined'
  | 'RecoveryInProgress'
  | 'ConsequenceManagement'
  | 'CorrectiveActionPending'
  | 'Closed';

export interface CaseDto {
  id: string;
  caseNumber: string;
  caseType: CaseType;
  status: CaseStatus;
  orgUnitId: string;
  title: string;
  description: string | null;
  amountInvolved: number | null;
  recoverableAmount: number;
  recoveredAmount: number;
  createdAtUtc: string;
}

export interface CreateCaseRequest {
  caseType: CaseType;
  orgUnitId: string;
  title: string;
  description: string | null;
  amountInvolved: number | null;
  supplierId?: string | null;
  contractId?: string | null;
}

export interface ChangeCaseStatusRequest {
  newStatus: CaseStatus;
  reason: string;
}

